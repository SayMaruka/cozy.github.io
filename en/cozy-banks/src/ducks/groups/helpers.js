import { groupBy, sortBy, deburr } from 'lodash'
import { ACCOUNT_DOCTYPE, GROUP_DOCTYPE } from 'doctypes'
import { associateDocuments } from 'ducks/client/utils'
import { getAccountType } from 'ducks/account/helpers'
import flag from 'cozy-flags'

export const getGroupLabel = (group, t) => {
  if (group.virtual) {
    return (
      t(`Data.accountTypes.${group.label}`, { _: 'other' }) +
      (flag('debug-groups') ? ' (virtual)' : '')
    )
  } else if (isAutoGroup(group) && !isFormerAutoGroup(group)) {
    return (
      t(`Data.accountTypes.${group.accountType}`) +
      (flag('debug-groups') ? ' (auto)' : '')
    )
  } else {
    return group.label
  }
}

export const buildAutoGroup = (accountType, accounts, options = {}) => {
  const { virtual = true, client = null } = options

  const group = {
    _type: GROUP_DOCTYPE,
    label: accountType,
    accountType: accountType
  }

  if (virtual) {
    group.virtual = true
    group._id = accountType
  }

  associateDocuments(group, 'accounts', ACCOUNT_DOCTYPE, accounts)

  if (client) {
    group.accounts = accounts.map(x => x._id)
    return client.hydrateDocument(group)
  } else {
    associateDocuments(group, 'accounts', ACCOUNT_DOCTYPE, accounts)
    return group
  }
}

export const buildAutoGroups = (accounts, options) => {
  const accountsByType = groupBy(accounts, getAccountType)

  const virtualGroups = Object.entries(accountsByType).map(
    ([accountType, accounts]) => buildAutoGroup(accountType, accounts, options)
  )

  return virtualGroups
}

/**
 * Returns a function that returns the translated label of a group
 *
 * @param {Object} group - Group
 * @param {Function} translate - Translation function
 * @returns {Object} Translated label
 */

const isOtherVirtualGroup = group => group.virtual && group.label === 'Other'

export const isReimbursementsVirtualGroup = group =>
  group.virtual && group._id === 'Reimbursements'

const getCategory = group => {
  if (isReimbursementsVirtualGroup(group)) {
    return 'virtualReimbursements'
  } else if (isOtherVirtualGroup(group)) {
    return 'virtualOther'
  } else {
    return 'normal'
  }
}

const groupSortingPriorities = {
  normal: 0,
  virtualOther: 1,
  virtualReimbursements: 2
}

/**
 * Translate groups labels then sort them on their translated label. But always put "others accounts" last
 * @param {Object[]} groups - The groups to sort
 * @param {Function} translate - The translation function
 * @returns {Object[]} The sorted wrapped groups ({ category, label, group })
 */
export const translateAndSortGroups = (groups, translate) => {
  // Wrap groups to add necessary information for sorting
  const wrappedGroups = groups.map(group => ({
    group,
    category: getCategory(group),
    label: getGroupLabel(group, translate)
  }))

  return sortBy(wrappedGroups, ({ label, category }) => [
    groupSortingPriorities[category],
    deburr(label).toLowerCase()
  ])
}

// For automatically created groups, the `accountType` attribute is present.
export const isFormerAutoGroup = group => group.accountType === null
export const isAutoGroup = group => group.accountType !== undefined
export const getGroupAccountType = group => group.accountType
