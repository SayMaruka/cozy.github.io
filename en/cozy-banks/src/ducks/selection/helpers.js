import GraphCircleIcon from 'cozy-ui/transpiled/react/Icons/GraphCircle'

export const makeSelectionBarActions = showTransactionCategoryModal => {
  return {
    categorize: {
      action: () => showTransactionCategoryModal(),
      displayCondition: selected => selected.length > 0,
      icon: GraphCircleIcon
    }
  }
}
