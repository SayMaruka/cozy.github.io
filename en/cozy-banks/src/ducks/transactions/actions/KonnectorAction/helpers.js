import { findMatchingBrand } from 'ducks/brandDictionary'

export function getBrandsWithoutTrigger(brands) {
  return brands.filter(brand => !brand.hasTrigger)
}

export const findMatchingBrandWithoutTrigger = (transaction, brands) => {
  const brandsWithoutTrigger = getBrandsWithoutTrigger(brands)

  if (!brandsWithoutTrigger) {
    return null
  }

  const matchingBrand = findMatchingBrand(
    brandsWithoutTrigger,
    transaction.label
  )

  if (!matchingBrand || matchingBrand.maintenance) {
    return null
  }

  return matchingBrand
}

export const hasUrls = urls => urls['COLLECT'] || urls['HOME']
