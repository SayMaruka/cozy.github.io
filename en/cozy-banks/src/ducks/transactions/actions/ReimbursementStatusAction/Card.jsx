import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'cozy-ui/react'
import UICard from 'cozy-ui/react/Card'
import { Caption, Text } from 'cozy-ui/react/Text'
import Icon, { iconPropType } from 'cozy-ui/react/Icon'
import cx from 'classnames'
import styles from 'ducks/transactions/actions/ReimbursementStatusAction/Card.styl'
import { getPlatform } from 'cozy-device-helper'

// TODO use icons from cozy-ui when https://github.com/cozy/cozy-ui/pull/983
// has been merged
import iconPhone from 'assets/icons/icon-phone.svg'
import iconEmail from 'assets/icons/icon-email.svg'

const Card = props => {
  const { title, caption, icon, className, ...rest } = props

  return (
    <UICard className={cx(styles.Card, className)} {...rest}>
      <Icon icon={icon} size={16} color="var(--coolGrey)" />
      <div className="u-ml-1">
        <Text>{title}</Text>
        <Caption>{caption}</Caption>
      </div>
    </UICard>
  )
}

Card.propTypes = {
  icon: iconPropType.isRequired,
  title: PropTypes.string.isRequired,
  caption: PropTypes.node.isRequired
}

export const DumbPhoneCard = props => {
  const { contact, t, ...rest } = props
  const href = `tel:${contact.number}`
  const caption = contact.price ? (
    <>
      {contact.price}
      {t('ReimbursementStatusModal.contact.phoneCallPrice')}
    </>
  ) : (
    t('ReimbursementStatusModal.contact.phoneCallFree')
  )

  return (
    <Card
      href={href}
      title={contact.number}
      caption={caption}
      icon={iconPhone}
      tag="a"
      {...rest}
    />
  )
}

const phoneContactShape = PropTypes.shape({
  price: PropTypes.string,
  number: PropTypes.string.isRequired
})

DumbPhoneCard.propTypes = {
  contact: phoneContactShape.isRequired,
  t: PropTypes.func.isRequired
}

export const PhoneCard = translate()(DumbPhoneCard)

export const DumbWebCard = props => {
  const { contact, t, ...rest } = props
  const caption = new URL(contact.href).hostname

  return (
    <Card
      href={contact.href}
      title={t(`ReimbursementStatusModal.contact.actions.${contact.action}`)}
      caption={caption}
      icon={iconEmail}
      target="_blank"
      tag="a"
      {...rest}
    />
  )
}

const webContactShape = PropTypes.shape({
  href: PropTypes.string.isRequired,
  action: PropTypes.string.isRequired
})

DumbWebCard.propTypes = {
  contact: webContactShape.isRequired,
  t: PropTypes.func.isRequired
}

export const WebCard = translate()(DumbWebCard)

export const DumbAppCard = props => {
  const { contact, t, ...rest } = props

  if (getPlatform() !== contact.platform) {
    return null
  }

  return (
    <Card
      href={contact.href}
      title={t('ReimbursementStatusModal.contact.openApp.title')}
      caption={t('ReimbursementStatusModal.contact.openApp.caption')}
      icon="phone"
      target="_blank"
      tag="a"
      {...rest}
    />
  )
}

const appContactShape = PropTypes.shape({
  platform: PropTypes.oneOf(['android', 'ios']).isRequired,
  href: PropTypes.string.isRequired
})

DumbAppCard.propTypes = {
  contact: appContactShape.isRequired,
  t: PropTypes.func.isRequired
}

export const AppCard = translate()(DumbAppCard)

export const ContactCard = props => {
  const { type, ...rest } = props

  switch (type) {
    case 'phone':
      return <PhoneCard {...rest} />

    case 'web':
      return <WebCard {...rest} />

    case 'app':
      return <AppCard {...rest} />

    default:
      return null
  }
}

ContactCard.propTypes = {
  type: PropTypes.oneOf(['phone', 'web', 'app']).isRequired,
  contact: PropTypes.object.isRequired
}
