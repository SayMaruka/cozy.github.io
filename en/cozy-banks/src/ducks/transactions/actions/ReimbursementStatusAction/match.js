import flag from 'cozy-flags'
import { isExpense } from 'ducks/transactions/helpers'

const match = transaction => {
  // TODO match only if the component is going to render something
  return isExpense(transaction) && flag('reimbursement-tag')
}

export default match
