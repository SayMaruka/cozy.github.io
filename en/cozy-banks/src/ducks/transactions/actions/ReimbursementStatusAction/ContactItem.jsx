import React from 'react'
import PropTypes from 'prop-types'
import { SubTitle } from 'cozy-ui/react/Text'
import cx from 'classnames'
import { ContactCard } from 'ducks/transactions/actions/ReimbursementStatusAction/Card'

const ContactItem = props => {
  const { brand, ...rest } = props

  return (
    <div {...rest}>
      <SubTitle className="u-mb-half">Contacter {brand.name}</SubTitle>
      {brand.contact.map((contact, index) => {
        const isLast = index === brand.contact.length - 1

        return (
          <ContactCard
            key={index}
            className={cx({ 'u-mb-half': !isLast })}
            type={contact.type}
            contact={contact}
          />
        )
      })}
    </div>
  )
}

ContactItem.propTypes = {
  brand: PropTypes.shape({
    name: PropTypes.string.isRequired,
    contact: PropTypes.arrayOf(PropTypes.object).isRequired
  })
}

export default ContactItem
