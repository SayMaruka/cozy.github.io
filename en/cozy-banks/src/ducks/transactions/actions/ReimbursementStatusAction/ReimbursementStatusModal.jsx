import React from 'react'
import { flowRight as compose } from 'lodash'
import { translate, withBreakpoints } from 'cozy-ui/react'
import Modal, { ModalContent } from 'cozy-ui/react/Modal'
import Icon from 'cozy-ui/react/Icon'
import { Title, Text } from 'cozy-ui/react/Text'
import cx from 'classnames'
import { List, Row, Radio } from 'components/List'
import iconReimbursement from 'assets/icons/icon-reimbursement-detailed.svg'
import styles from 'ducks/transactions/actions/ReimbursementStatusAction/ReimbursementStatusModal.styl'
import { getReimbursementStatus, getLabel } from 'ducks/transactions/helpers'
import { isHealthExpense } from 'ducks/categories/helpers'
import flag from 'cozy-flags'
import ContactItem from 'ducks/transactions/actions/ReimbursementStatusAction/ContactItem'

class _ReimbursementStatusModal extends React.PureComponent {
  render() {
    const {
      transaction,
      onChange,
      t,
      className,
      breakpoints: { isMobile },
      brands,
      ...rest
    } = this.props
    const choices = ['pending', 'reimbursed', 'no-reimbursement']
    const status = getReimbursementStatus(transaction)

    const showContacts =
      flag('reimbursements-contacts') && isHealthExpense(transaction)

    return (
      <Modal
        mobileFullscreen
        className={cx('u-flex', 'u-flex-column', className)}
        {...rest}
      >
        <ModalContent className="u-ph-0">
          <header className="u-ta-center">
            <Icon icon={iconReimbursement} size={56} color="var(--slateGrey)" />
            <Title className="u-mt-1">
              {t('Transactions.actions.reimbursementStatus.modal.title')}
            </Title>
            <Text className={styles.ReimbursementStatusModal__transactionLabel}>
              {getLabel(transaction)}
            </Text>
          </header>
          <form className="u-mt-1">
            <List border="vertical">
              {choices.map(choice => (
                <Row key={choice}>
                  <Radio
                    key={choice}
                    name="reimbursementStatus"
                    value={choice}
                    label={t(`Transactions.reimbursementStatus.${choice}`)}
                    checked={status === choice}
                    onChange={onChange}
                    className={cx('u-mb-0', styles.Radio)}
                  />
                </Row>
              ))}
            </List>
          </form>
          {showContacts ? (
            <div
              className={cx(
                styles.ReimbursementStatusModal__contact,
                'u-pt-2',
                {
                  'u-mt-auto': isMobile
                }
              )}
            >
              {brands
                .filter(
                  brand => brand.health && brand.hasTrigger && brand.contact
                )
                .map((brand, index) => (
                  <ContactItem
                    brand={brand}
                    key={brand.name}
                    // TODO use stack layout when https://github.com/cozy/cozy-banks/pull/1312 has been merged (see https://github.com/cozy/cozy-banks/pull/1312/commits/2bc1d75a25fe2c61f219579ac56407e356997105 more particularly)
                    className={cx({
                      'u-mt-1-half': index !== 0
                    })}
                  />
                ))}
            </div>
          ) : null}
        </ModalContent>
      </Modal>
    )
  }
}

const ReimbursementStatusModal = compose(
  translate(),
  withBreakpoints()
)(_ReimbursementStatusModal)

export default ReimbursementStatusModal
