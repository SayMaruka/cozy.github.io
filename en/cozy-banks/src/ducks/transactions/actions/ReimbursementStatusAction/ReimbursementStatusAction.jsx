import React from 'react'
import { connect } from 'react-redux'
import Icon from 'cozy-ui/transpiled/react/Icon'
import Chip from 'cozy-ui/transpiled/react/Chip'
import Alerter from 'cozy-ui/transpiled/react/Alerter'
import {
  getReimbursementStatus,
  isReimbursementLate,
  REIMBURSEMENTS_STATUS
} from 'ducks/transactions/helpers'
import TransactionModalRow, {
  RowArrow
} from 'ducks/transactions/TransactionModalRow'
import ReimbursementStatusModal from 'ducks/transactions/actions/ReimbursementStatusAction/ReimbursementStatusModal'
import iconReimbursement from 'assets/icons/icon-reimbursement.svg'
import { logException } from 'lib/sentry'
import cx from 'classnames'
import { translate } from 'cozy-ui/transpiled/react'
import { flowRight as compose } from 'lodash'
import { withMutations } from 'cozy-client'
import { getHealthReimbursementLateLimitSelector } from 'ducks/reimbursements/selectors'

export class DumbReimbursementStatusAction extends React.PureComponent {
  state = {
    showModal: false
  }

  showModal = () => this.setState({ showModal: true })
  hideModal = () => this.setState({ showModal: false })

  handleChange = async e => {
    const { transaction, saveDocument, t } = this.props
    transaction.reimbursementStatus = e.target.value

    this.hideModal()

    try {
      await saveDocument(transaction)
    } catch (err) {
      logException(err)
      Alerter.error(t('Transactions.reimbursementStatusUpdateError'))
    }
  }

  renderModalItem() {
    const { t, transaction, healthReimbursementLateLimit } = this.props

    const status = getReimbursementStatus(transaction)
    const isLate = isReimbursementLate(
      transaction,
      healthReimbursementLateLimit
    )
    const translateKey = isLate ? 'late' : status
    const label = t(`Transactions.actions.reimbursementStatus.${translateKey}`)

    return (
      <TransactionModalRow
        iconLeft={<Icon icon={iconReimbursement} />}
        iconRight={<RowArrow />}
        onClick={this.showModal}
      >
        {label}
      </TransactionModalRow>
    )
  }

  renderTransactionRow() {
    const { transaction, t, healthReimbursementLateLimit } = this.props

    const status = getReimbursementStatus(transaction)
    const isLate = isReimbursementLate(
      transaction,
      healthReimbursementLateLimit
    )

    if (status === REIMBURSEMENTS_STATUS.noReimbursement) {
      return null
    }

    const translateKey = isLate ? 'late' : status

    return (
      <Chip
        size="small"
        variant="outlined"
        theme={isLate ? 'error' : 'normal'}
        onClick={this.showModal}
        className={cx({ 'u-valid': status === 'reimbursed' })}
      >
        {t(`Transactions.actions.reimbursementStatus.${translateKey}`)}
        {status === 'pending' && (
          <>
            <Chip.Separator />
            <Icon icon="hourglass" size={12} />
          </>
        )}
      </Chip>
    )
  }

  render() {
    const { isModalItem, transaction } = this.props

    return (
      <>
        {isModalItem ? this.renderModalItem() : this.renderTransactionRow()}
        {this.state.showModal && (
          <ReimbursementStatusModal
            into="body"
            dismissAction={this.hideModal}
            mobileFullscreen
            transaction={transaction}
            onChange={this.handleChange}
            brands={this.props.actionProps.brands}
          />
        )}
      </>
    )
  }
}

const ReimbursementStatusAction = compose(
  translate(),
  withMutations(),
  connect(state => ({
    healthReimbursementLateLimit: getHealthReimbursementLateLimitSelector(state)
  }))
)(DumbReimbursementStatusAction)

export default ReimbursementStatusAction
