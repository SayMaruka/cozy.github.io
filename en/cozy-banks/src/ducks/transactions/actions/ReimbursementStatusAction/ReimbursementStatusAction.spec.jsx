import { DumbReimbursementStatusAction } from './ReimbursementStatusAction'
import React from 'react'
import { shallow } from 'enzyme'
import { getReimbursementStatus, isReimbursementLate } from '../../helpers'

jest.mock('../../helpers')

describe('DumbReimbursementStatusAction', () => {
  const t = key => key

  describe('transaction row context', () => {
    it('should render correctly for a pending reimbursement', () => {
      getReimbursementStatus.mockReturnValueOnce('pending')

      const wrapper = shallow(
        <DumbReimbursementStatusAction
          t={t}
          transaction={{ reimbursementStatus: 'pending' }}
          isModalItem={false}
        />
      )

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render correctly for a reimbursed transaction', () => {
      getReimbursementStatus.mockReturnValueOnce('reimbursed')

      const wrapper = shallow(
        <DumbReimbursementStatusAction
          t={t}
          transaction={{ reimbursementStatus: 'reimbursed' }}
          isModalItem={false}
        />
      )

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render correctly for a late reimbursement', () => {
      getReimbursementStatus.mockReturnValueOnce('pending')
      isReimbursementLate.mockReturnValueOnce(true)

      const wrapper = shallow(
        <DumbReimbursementStatusAction
          t={t}
          transaction={{ reimbursementStatus: 'pending' }}
          isModalItem={false}
        />
      )

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render correctly for a transaction that will not be reimbursed', () => {
      getReimbursementStatus.mockReturnValueOnce('no-reimbursement')

      const wrapper = shallow(
        <DumbReimbursementStatusAction
          t={t}
          transaction={{ reimbursementStatus: 'no-reimbursement' }}
          isModalItem={false}
        />
      )

      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('modal item context', () => {
    it('should render correctly for a pending reimbursement', () => {
      getReimbursementStatus.mockReturnValueOnce('pending')

      const wrapper = shallow(
        <DumbReimbursementStatusAction
          t={t}
          transaction={{ reimbursementStatus: 'pending' }}
          isModalItem={true}
        />
      )

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render correctly for a reimbursed transaction', () => {
      getReimbursementStatus.mockReturnValueOnce('reimbursed')

      const wrapper = shallow(
        <DumbReimbursementStatusAction
          t={t}
          transaction={{ reimbursementStatus: 'reimbursed' }}
          isModalItem={true}
        />
      )

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render correctly for a late reimbursement', () => {
      getReimbursementStatus.mockReturnValueOnce('pending')
      isReimbursementLate.mockReturnValueOnce(true)

      const wrapper = shallow(
        <DumbReimbursementStatusAction
          t={t}
          transaction={{ reimbursementStatus: 'pending' }}
          isModalItem={true}
        />
      )

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render correctly for a transaction that will not be reimbursed', () => {
      getReimbursementStatus.mockReturnValueOnce('no-reimbursement')

      const wrapper = shallow(
        <DumbReimbursementStatusAction
          t={t}
          transaction={{ reimbursementStatus: 'no-reimbursement' }}
          isModalItem={true}
        />
      )

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
