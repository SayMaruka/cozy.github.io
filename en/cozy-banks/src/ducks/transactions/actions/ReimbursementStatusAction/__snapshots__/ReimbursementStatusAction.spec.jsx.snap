// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DumbReimbursementStatusAction modal item context should render correctly for a late reimbursement 1`] = `"<div class=\\"media TransactionModalRow TransactionModalRow-clickable u-pl-1-half\\"><div class=\\"img TransactionModalRowIcon\\"><svg class=\\"icon\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#test-file-stub\\"></use></svg></div><div class=\\"bd TransactionModalRowContent\\">Transactions.actions.reimbursementStatus.late</div></div>"`;

exports[`DumbReimbursementStatusAction modal item context should render correctly for a pending reimbursement 1`] = `"<div class=\\"media TransactionModalRow TransactionModalRow-clickable u-pl-1-half\\"><div class=\\"img TransactionModalRowIcon\\"><svg class=\\"icon\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#test-file-stub\\"></use></svg></div><div class=\\"bd TransactionModalRowContent\\">Transactions.actions.reimbursementStatus.pending</div></div>"`;

exports[`DumbReimbursementStatusAction modal item context should render correctly for a reimbursed transaction 1`] = `"<div class=\\"media TransactionModalRow TransactionModalRow-clickable u-pl-1-half\\"><div class=\\"img TransactionModalRowIcon\\"><svg class=\\"icon\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#test-file-stub\\"></use></svg></div><div class=\\"bd TransactionModalRowContent\\">Transactions.actions.reimbursementStatus.reimbursed</div></div>"`;

exports[`DumbReimbursementStatusAction modal item context should render correctly for a transaction that will not be reimbursed 1`] = `"<div class=\\"media TransactionModalRow TransactionModalRow-clickable u-pl-1-half\\"><div class=\\"img TransactionModalRowIcon\\"><svg class=\\"icon\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#test-file-stub\\"></use></svg></div><div class=\\"bd TransactionModalRowContent\\">Transactions.actions.reimbursementStatus.no-reimbursement</div></div>"`;

exports[`DumbReimbursementStatusAction transaction row context should render correctly for a late reimbursement 1`] = `"<div class=\\"c-chip c-chip--smallSize c-chip--outlinedVariant c-chip--errorTheme u-breakword c-chip--hoverableErrorTheme c-chip--clickable\\">Transactions.actions.reimbursementStatus.late<span class=\\"c-chip-separator\\"></span><svg class=\\"icon\\" width=\\"12\\" height=\\"12\\"><use xlink:href=\\"#hourglass\\"></use></svg></div>"`;

exports[`DumbReimbursementStatusAction transaction row context should render correctly for a pending reimbursement 1`] = `"<div class=\\"c-chip c-chip--smallSize c-chip--outlinedVariant c-chip--normalTheme u-breakword c-chip--outlinedNormalTheme c-chip--hoverableNormalTheme c-chip--clickable\\">Transactions.actions.reimbursementStatus.pending<span class=\\"c-chip-separator\\"></span><svg class=\\"icon\\" width=\\"12\\" height=\\"12\\"><use xlink:href=\\"#hourglass\\"></use></svg></div>"`;

exports[`DumbReimbursementStatusAction transaction row context should render correctly for a reimbursed transaction 1`] = `"<div class=\\"c-chip c-chip--smallSize c-chip--outlinedVariant c-chip--normalTheme u-breakword c-chip--outlinedNormalTheme c-chip--hoverableNormalTheme c-chip--clickable u-valid\\">Transactions.actions.reimbursementStatus.reimbursed</div>"`;

exports[`DumbReimbursementStatusAction transaction row context should render correctly for a transaction that will not be reimbursed 1`] = `""`;
